simple_certmanager/models.py

```diff
@@ -10,12 +10,76 @@
 )
 from privates.fields import PrivateMediaFileField
 
+from simple_certmanager.csr_generation import generate_private_key_with_csr
+
 from .constants import CertificateTypes
 from .mixins import DeleteFileFieldFilesMixin
 from .utils import load_pem_x509_private_key, pretty_print_certificate_components
 from .validators import PrivateKeyValidator, PublicCertValidator
 
 
+class SigningRequest(models.Model):
+    common_name = models.CharField(
+        max_length=100,
+        help_text=_("The common name for the certificate"),
+    )
+    country_name = models.CharField(
+        max_length=2, help_text=_("Two-letter country code"), blank=True, default="NL"
+    )
+    organization_name = models.CharField(
+        max_length=100, help_text=_("The name of the organization"), blank=True
+    )
+    state_or_province_name = models.CharField(
+        max_length=100, help_text=_("The state or province name"), blank=True
+    )
+    locality_name = models.CharField(
+        max_length=100, help_text=_("The locality name"), blank=True
+    )
+    email_address = models.EmailField(
+        help_text=_("Email address for the certificate"), blank=True
+    )
+    csr = models.TextField(
+        editable=False,
+        blank=True,
+        help_text=_("Certificate Signing Request"),
+        verbose_name=_("CSR"),
+    )
+    private_key = models.TextField(
+        editable=False,
+        blank=True,
+    )
+
+    class Meta:
+        constraints = [
+            models.CheckConstraint(
+                check=models.Q(
+                    models.Q(csr__isnull=False, private_key__isnull=False)
+                    | models.Q(csr__isnull=True, private_key__isnull=True),
+                ),
```

Comment:
ah this isn't entirely it - because they are text fields, they will never be `null` (unless you set `null=True` on the model fields, but we specifically do not allow this on text fields)

The correct check would be something like:

```py
models.Q(csr="", private_key="") 
| (~models.Q(csr="") & ~models.Q(private_key=""))
```